### PR TITLE
Client refactor & enhancement

### PR DESCRIPTION
**Pull request by furiousdom**
This PR refactors & configures client in general.

### CODE REVIEW

File: client/src/components/Note.vue [hunk 1]
```diff
@@ -7,33 +7,33 @@
...
+    <p v-if="error" class="error">{{ error }}</p>
     <div class="columns is-flex-tablet">
-      <div class="column"
-      v-for="(note, index) in notes"
-      v-bind:item="note"
-      v-bind:index="index"
-      v-bind:key="note._id"
-      >
+      <div
+        v-for="(note, index) in notes"
```
Comment (lukasaric) [inline client/src/components/Note.vue]: Destructure directly `_id`, `title`, `text` and `createdAt`. You don't need index.

File: client/src/components/Note.vue [hunk 2]
```diff
@@ -7,33 +7,33 @@
...
+        :key="note._id"
+        :item="note"
```
Comment (lukasaric) [inline client/src/components/Note.vue]: Why binding? Remove the `item` and `index` binding

File: client/src/api/note.js [hunk 3]
```diff
@@ -32,7 +31,6 @@ class NoteService {
   static deleteNote(id) {
     return axios.delete(`${url}${id}`);
   }
-
 }
 
 export default NoteService;
```
Comment (lukasaric) [inline client/src/api/note.js]: See structure in [phonebook](https://github.com/lukasaric/Phonebook/tree/master/client/src/services) for starters. Refactor it, then we will do some more refactoring.
- Name api file in phonebook here as `axios.js`.

File: client/src/components/Note.vue [hunk 4]
```diff
@@ -46,28 +46,21 @@
...
-  async created() {
-    try {
-      this.notes = await NoteService.getNotes();
-    } catch (err) {
-      this.error = err.message;
-    }
+    };
   },
   methods: {
     async createNote() {
```
[the remark this answers is not in the document]
Comment (furiousdom) [reply client/src/components/Note.vue]: This we will need to talk about over the phone...

File: client/src/api/axios.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import axios from 'axios';
+
+const axiosInstance = axios.create({
+  headers: { 'Content-Type': 'application/json' },
```
Comment (lukasaric) [inline client/src/api/axios.js:4]: Extract options on top. Name it `config`.

File: client/src/api/note.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import api from './axios';
+const endpoint = 'notes/';
+
+export default {
```
Comment (lukasaric) [inline client/src/api/note.js:4]: Extract all methods as functions on top, then export it:
```js
export default { fetch, create ... }
```

File: client/src/api/note.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  getNotes() {
```
Comment (lukasaric) [inline client/src/api/note.js:5]: Name it `fetch`.

File: client/src/api/note.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import api from './axios';
+const endpoint = 'notes/';
```
Comment (lukasaric) [inline client/src/api/note.js:2]: Create an object and name it `url`, with `root` prop as `/notes`.

File: client/src/api/note.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+export default {
+  getNotes() {
+    /* eslint-disable no-async-promise-executor */
+    return new Promise(async (resolve, reject) => {
```
Comment (lukasaric) [inline client/src/api/note.js:7]: You don't need to do this use `new Promise and remove try-catch as well.
Just do this:
```js
function fetch() {
	const { data } = await api.get(url.root);
	return data.map(note => ({ ...note, createdAt: new Date(note.createdAt) });
}
	
```

File: client/src/api/note.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+            ...note,
+            createdAt: new Date(note.createdAt)
+          }))
+        );
+      } catch (err) {
+        reject(err);
+      }
+    });
+  },
+  insertNote(title, text) {
```
Comment (lukasaric) [inline client/src/api/note.js:21]: - Name it `create`.
- Accept one object argument with `title` and `text` in it. Name it `data` or `item`

File: client/src/api/note.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    return api.post(endpoint, { title, text });
```
Comment (lukasaric) [inline client/src/api/note.js:22]:
```js
return api.post(url.root, item);
```

File: client/src/components/Note.vue [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+              <input v-model="text" type="text" placeholder="Take a note..." class="input">
+            </div>
+          </div>
+          <div class="field">
+            <button @click="addNote" class="button">Save</button>
+          </div>
+        </div>
+      </div>
+    </div>
+
```
Comment (lukasaric) [inline client/src/components/Note.vue:24]: Remove empty spaces everywhere in template.

File: client/src/components/Note.vue [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+    <hr>
+
+    <p v-if="error" class="error">{{ error }}</p>
+    <div class="columns is-flex-tablet">
+      <div
+        v-for="({ _id, title, text, createdAt:date }) in notes"
```
Comment (lukasaric) [inline client/src/components/Note.vue:30]: Indent `date` alias.

File: client/src/components/Note.vue [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+      const newNote = {
+        _id,
+        title,
+        text,
+        createdAt: new Date(date)
+      };
+      return newNote;
+    },
+    async addNote() {
+      const { data } = await NoteServices.insertNote(this.title, this.text);
```
Comment (lukasaric) [inline client/src/components/Note.vue:70]:
```js
const { title, text } = this;
const { data } = await NoteServices.insertNote({ title, text });
```

**Pull request merged**